### PR TITLE
Colour by Property - bug fixes, enhancements, updates

### PR DESCRIPTION
**Pull request by lilyclements**
@africanmathsinitiative/developers this is ready to review

### CODE REVIEW

File: instat/dlgColourbyProperty.vb [hunk 1]
```diff
@@ -17,44 +17,59 @@
...
-        ucrReceiverMetadataProperty.SetParameter(New RParameter("property"))
-        ucrReceiverMetadataProperty.SetParameterIsString()
+    Private Sub SetDefaults()
+        clsColourByMetadata = New RFunction
+        clsRemoveColour = New RFunction
 
-        ucrSelectorColourByMetadata.SetParameter(New RParameter("data_name"))
-        ucrSelectorColourByMetadata.SetParameterIsString()
+        ucrSelectorColourByMetadata.Reset()
+        ucrChkRemoveColours.Checked = False
```
Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: @africanmathsinitiative/developers This is not right. Do we have a way to do something like an `AddAdditionalPairParameter` but for a function in the `SetRCodeForControls`?

Comment (dannyparsons) [inline instat/dlgColourbyProperty.vb]: The conditions set above should work if you set the R code for the checkbox below.

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: I have two classes: One is for when the checkbox is checked and the other is for when it isn't checked.
If I `SetRCode` for one class, then the checkbox is always checked
If I `SetRCode` for the other class, then the checkbox is always unchecked.

Comment (dannyparsons) [inline instat/dlgColourbyProperty.vb]: Do you set one as the base function? If so, the R code for the check box as the base function.

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: Found the bug. I was calling `SetRCodeForControls(False)` at one point. Fixed now!

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: Thanks

File: instat/dlgColourbyProperty.vb [hunk 2]
```diff
@@ -17,44 +17,60 @@
...
-    Private Sub SetDefaults()
-        '  SetOrRemoveColours()
-        ucrBase.clsRsyntax.SetBaseRFunction(clsDefaultFuncAdd.Clone())
-        ucrSelectorColourByMetadata.Reset()
-        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, True)
+    Private Sub SetRCodeForControls(bReset As Boolean)
+        ucrSelectorColourByMetadata.AddAdditionalCodeParameterPair(clsRemoveColour, ucrSelectorColourByMetadata.GetParameter(), iAdditionalPairNo:=1)
+        ucrSelectorColourByMetadata.SetRCode(clsColourByMetadata, bReset)
+        ucrReceiverMetadataProperty.SetRCode(clsColourByMetadata, bReset)
+        ucrChkRemoveColours.SetRCode(clsColourByMetadata, bReset)
```
Comment (dannyparsons) [inline instat/dlgColourbyProperty.vb]: Isn't this still always going to be unchecked?

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: No, it allows me to check and uncheck it now, and is default unchecked (as we want, I assume? It was unchecked by default beforehand!)

Comment (dannyparsons) [inline instat/dlgColourbyProperty.vb]: If you check then close and reopen the dialog does it stay checked?

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: Sorcery. How did you know it would do that?
What have I done wrong?

Comment (dannyparsons) [inline instat/dlgColourbyProperty.vb]: As you said:
> If I SetRCode for the other class, then the checkbox is always unchecked.

So in SetRCode you need to set the RCode for the checkbox to be the base function, so it's always the correct function it's checking.

Comment (lilyclements) [inline instat/dlgColourbyProperty.vb]: Well aint that fancy. Done that now, thanks!

**Comment by lilyclements** [comment]
@africanmathsinitiative/developers this is ready to review

**Pull request merged**